chainer/functions/math/sparse_matmul.py

```diff
@@ -0,0 +1,429 @@
+import numpy
+
+import chainer
+from chainer.backends import cuda
+from chainer import function_node
+from chainer import utils
+from chainer.utils import type_check
+
+try:
+    from scipy import sparse
+    _scipy_available = True
+except ImportError:
+    _scipy_available = False
+
+
+def _sparse_matmul(sp_data, sp_row, sp_col, sp_shape, dn,
+                   transa, transb, transc, dtype=None):
```

Comment:
Just a comment:
Should it be `_sparse_matmul_coo` or something by considering that the other type of sparse_matmul (matmul with CSR matrix etc.) will be introduced in the future?

Same applies for all methods/class except `sparse_matmul`. 
I think it is ok to keep `sparse_matmul`, since it is checking the class type to see which function to be used.

Reply:
I agree we can be more specific here, although we can probably come back and fix these later considering them implementation details. @anaruse Could you please revisit the names of these functions?

Reply:
Thanks for your comments, @corochann and @hvy.
So how about renaming them as follows (while keeping `sparse_matmul`)?
 * _sparse_matmul* --> _coo_matmul*
 * SparseMatMul* --> CooMatMul*

Reply:
Should we be specific about both the LHS and RHS, e,g. `CooDenseMatMul`?

Reply:
AFAIK, matrix multiplication of sparse vs sparse is not so common. So, I don't think we should be specific like that.

Reply:
I see, it might be clear from the arguments too. Could you change the names to your above suggestions?

Reply:
Oh, I'm sorry, I missed your comment above. So, I will change the names in that way.